cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -0,0 +1,163 @@
+    Assert.assertTrue(client.getApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, false).isEmpty());
+    Assert.assertEquals(propMap, client.getApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, true));
+    Assert.assertEquals(propMap, client.getProgramPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, "flows",
+                                                              FakeApp.FLOWS.get(0), true));
+
+    client.deleteNamespacePreferences(Constants.DEFAULT_NAMESPACE);
+    propMap.put("k1", "instance");
+    Assert.assertTrue(client.getNamespacePreferences(Constants.DEFAULT_NAMESPACE, false).isEmpty());
+    Assert.assertEquals(propMap, client.getNamespacePreferences(Constants.DEFAULT_NAMESPACE, true));
+    Assert.assertEquals(propMap, client.getApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, true));
+    Assert.assertEquals(propMap, client.getProgramPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, "flows",
+                                                              FakeApp.FLOWS.get(0), true));
+
+    client.deleteInstancePreferences();
+    propMap.clear();
+    Assert.assertEquals(propMap, client.getInstancePreferences());
+    Assert.assertEquals(propMap, client.getNamespacePreferences(Constants.DEFAULT_NAMESPACE, true));
+    Assert.assertEquals(propMap, client.getNamespacePreferences(Constants.DEFAULT_NAMESPACE, true));
+    Assert.assertEquals(propMap, client.getApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, true));
+    Assert.assertEquals(propMap, client.getProgramPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, "flows",
+                                                              FakeApp.FLOWS.get(0), true));
+    appClient.delete(FakeApp.NAME);
+    namespaceClient.delete(invalidNamespace);
+  }
+
+  @Test
+  public void testDeletingApp() throws Exception {
+    Map<String, String> propMap = Maps.newHashMap();
+    File jarFile = createAppJarFile(FakeApp.class);
+    appClient.deploy(jarFile);
+
+    propMap.put("k1", "application");
+    client.setApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, propMap);
+    Assert.assertEquals(propMap, client.getApplicationPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, false));
+
+    propMap.put("k1", "program");
+    client.setProgramPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, "flows", FakeApp.FLOWS.get(0), propMap);
+    Assert.assertEquals(propMap, client.getProgramPreferences(Constants.DEFAULT_NAMESPACE, FakeApp.NAME, "flows",
+                                                              FakeApp.FLOWS.get(0), false));
+
+    appClient.delete(FakeApp.NAME);
+    // deleting the app should have deleted the preferences that were stored. so deploy the app and check
+    // if the preferences are empty. we need to deploy the app again since getting preferences of non-existent apps
+    // is not allowed by the API.
+    appClient.deploy(jarFile);
```

Comment:
Would it be possible to merge this test into the previous test? Might save a significant amount of time, since you are already deleting the app in the previous test.